exploration/validate_outputs.md

```diff
@@ -0,0 +1,263 @@
+---
+jupyter:
+  jupytext:
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.3'
+      jupytext_version: 1.16.3
+  kernelspec:
+    display_name: venv
+    language: python
+    name: python3
+---
+
+## Validate output stats
+
+In this notebook we're taking a look at our output raster stats and comparing against the outputs from other raster stats calculation functions. We would expect our outputs to be the same as the `zonal_stats` outputs, and see some moderate differences with the `exactextract` outputs.
+
+```python
+%load_ext jupyter_black
+%load_ext autoreload
+%autoreload 2
+```
+
+```python
+from exactextract import exact_extract
+from rasterstats import zonal_stats
+from dotenv import load_dotenv
+from sqlalchemy import create_engine, text
+import os
+import pandas as pd
+import geopandas as gpd
+import rioxarray as rxr
+import numpy as np
+
+# os.chdir("..")
+from src.utils.cloud_utils import get_cog_url
```

Comment:
maybe something weird happening here w/ the methods we are mixing to import local modules... getting the error:

```
{
	"name": "ImportError",
	"message": "cannot import name 'get_cog_url' from 'src.utils.cloud_utils' (/Users/zackarno/Documents/CHD/repos/hdx-floodscan/src/utils/cloud_utils.py)",
	"stack": "---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Cell In[5], line 12
      9 import numpy as np
     11 # os.chdir(\"..\")
---> 12 from src.utils.cloud_utils import get_cog_url
     13 from src.utils.raster_utils import upsample_raster, prep_raster
     15 load_dotenv()

ImportError: cannot import name 'get_cog_url' from 'src.utils.cloud_utils' (/Users/zackarno/Documents/CHD/repos/hdx-floodscan/src/utils/cloud_utils.py)"
}
```

Reply:
Oh sorry that's on me for not making these fully reproducible! I put these up as some quick-and-dirty notebooks and didn't put a ton of thought into the setup there. The `os.chdir("..")` is just my hack for making these work since I don't love the overhead of requiring `pyproject.toml`/`setup.cfg` files and doing the install.

Reply:
no worries! i hear you on the `pyproject.toml`/`setup.cfg` overhead point - but I guess good if python team can choose a nice standard way to deal with it.  I think if the files were just generated and git tracked there would not really be any overhead... 

i believe some of this was handled by the cookiecutter repo (which you helped set up long ago!), however cookiecutter might be outdated now as i think it also gives you a bunch of anticipy configs which we don't necessarily want?  Also perhaps the files were added to gitignore instead of tracked? I think gitignoring them is kind of annoying because if someone clones your repo then they don't have them!

Reply:
Yes I agree that if we're using that setup then those files should definitely be tracked. It's a great point to think about standard setups as our different kinds of python projects change -- eg. between pipelines and AA projects and dashboards. I remember the cookiecutter was set up in the context of AA projects so there are probably a number of things that don't really apply to pipelines. I think notebooks are a good example of how some of these use cases are different. For example, in AA projects we usually use notebooks as "the" output, so it makes sense we have a good repo structure around that. In this repo, I'm (perhaps a bit lazily..) less concerned with notebook setup since the main pipelines code is the main output here. 

Although not a great excuse since it's still important to have others able to run the code -- especially when I ask them to review it! But at least in the former case with AA projects I think it's absolutely important to have a robust setup, whereas here I'm more comfortable with a bit more of a lightweight hack (eg. using `os.chdir(..)` to manage dependencies).